src/bin/ipc_latency.rs

```diff
@@ -88,18 +134,45 @@ macro_rules! netlink_bench {
                 .output()
                 .expect("make failed to start");
 
-            let (tx, rx) = mpsc::channel::<Vec<Duration>>();
+            let (tx, rx) = mpsc::channel::<Vec<(Duration, Duration, Duration)>>();
 
             // listen
             let c1 = thread::spawn(move || {
                 let mut buf = [0u8; 1024];
-                let nl = portus::ipc::netlink::Socket::<$mode>::new()
-                    .map(|sk| {
-                        Backend::new(sk, Arc::new(atomic::AtomicBool::new(true)), &mut buf[..])
-                    })
+                // What does the backend represent?
+                let mut nl = portus::ipc::netlink::Socket::<$mode>::new()
+                    .map(|sk| Backend::new(sk, Arc::new(atomic::AtomicBool::new(true)), &mut buf[..]))
                     .expect("nl ipc initialization");
                 tx.send(vec![]).expect("ok to insmod");
-                tx.send(bench(&nl.sender(), nl, iter)).expect("report rtts");
+                // this is what I added
+                nl.next().expect("receive echo");
+
+                let sender = nl.sender();
+                let res  = (0..iter)
+                .map(|_| {
+
+                    let portus_send_time= time::get_time(); 
+                    let mut portus_recv_time = time::get_time();
+                    let mut kern_recv_time = time::get_time();
+                    let mut kern_send_time = time::get_time();
+                    let msg = portus::serialize::serialize(&TimeMsg(portus_send_time)).expect("serialize");
+
+                    sender.send_msg(&msg[..]).expect("send ts");
+                    if let portus::serialize::Msg::Other(raw) =
+                        nl.next().expect("recv echo")
+                    {
+                        portus_recv_time = time::get_time();
+                        let kern_recv_msg = DetTimeMsg::from_raw_msg(raw).expect("get time from raw");
+                        kern_recv_time = kern_recv_msg.0;
+                        kern_send_time = kern_recv_msg.1;
+
+                    } else {
+                        panic!("wrong type");
+                    }
+                    return (portus_recv_time-portus_send_time, kern_recv_time-portus_send_time, portus_recv_time-kern_send_time)
```

Comment:
spaces between infix operators please, and should probably split this into 3 lines